### PR TITLE
Package liburing

### PR DESCRIPTION
**Pull request by davidbrochart**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by davidbrochart** [comment]
@conda-forge/help-c-cpp this is ready for review.

File: recipes/liburing/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - make
+
+test:
+  commands:
+    - test -f "$PREFIX/include/liburing.h"
+    - test -f "$PREFIX/lib/liburing.a"
```
Comment (xhochy) [inline recipes/liburing/meta.yaml:25]: Static libs should not be in the main package 
```suggestion
    - test ! -f "$PREFIX/lib/liburing.a"
```

File: recipes/liburing/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    - test -f "$PREFIX/lib/liburing.so"
+    - test -f "$PREFIX/lib/pkgconfig/liburing.pc"
+    - test -f "$PREFIX/lib/liburing-ffi.a"
```
Comment (xhochy) [inline recipes/liburing/meta.yaml:28]: Same here

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/liburing/meta.yaml (hunk 1 above)
Comment (davidbrochart) [reply recipes/liburing/meta.yaml:25]: Thanks, I took inspiration from [libuv-feedstock](https://github.com/conda-forge/libuv-feedstock/blob/25d64e3244ede518b02a546e4c262dd103df78e8/recipe/meta.yaml#L33), do they do it wrong there too?

Comment (xhochy) [reply recipes/liburing/meta.yaml]: Yes. That is also wrong.

File: recipes/liburing/build.sh [hunk 3]
```diff
@@ -0,0 +1,4 @@
+./configure --cc="$GCC" --cxx="$GXX" --prefix="$PREFIX"
```
Comment (xhochy) [inline recipes/liburing/build.sh:1]: Can you try whether `--disable-static` works? That would be a better alternative to creating and then instantly deleting the static archives.

Comment (davidbrochart) [reply recipes/liburing/build.sh:1]: I tried in https://github.com/conda-forge/staged-recipes/pull/25597/commits/1d6411ca2620ae0c948fe861892eeaedd14983ce, but there is no such option.

File: recipes/liburing/meta.yaml (hunk 1 above)
Comment (davidbrochart) [reply recipes/liburing/meta.yaml]: Indeed, they have an issue: https://github.com/conda-forge/libuv-feedstock/issues/63.

**Review by xhochy — APPROVED** [review]

**Pull request merged**